File: problems/Two Pointers/TwoSumII.js
var twoSum = function (numbers, target) {
	let l = 1;
	var n = numbers.length;
	let r = n;

	while (l < r) {
		if (target == numbers[l] + numbers[r]) {
			return [l + 1, r + 1];
		} else if (target > numbers[l] + numbers[r]) {
			l++;
		} else if (target < numbers[l] + numbers[r]) {
			r--;
		}
	}
};

console.log(twoSum([2, 7, 11, 15], 9));

File: problems/Two Pointers/TwoSumII.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        l = 0
        r = n - 1

        while l < r:
            sum = numbers[l] + numbers[r]
            if target == sum:
                return [l+1, r+1]
            elif target > sum:
                l += 1
            elif target < sum:
                r -= 1
